torch/utils/checkpoint.py

```diff
@@ -367,7 +371,7 @@ def inner_unpack(packed):
                     with torch.autograd.graph.saved_tensors_hooks(inner_pack, inner_unpack):
                         _unused = function(*args)
 
-        return storage[x]
+        return storage.pop(x)
```

Comment:
We should properly check if the value is there or if it was already removed here.

Reply:
@albanD to be clear, this function should always return a tensor, so if it is not, we should raise an error?

Reply:
If you can't return the original Tensor, then yes it should raise an error.
That being said, we should try to always make this function work.

Reply:
Sure, I think the case that you mentioned works, are there any other edge cases you can think of that we'd like to test?

Reply:
yeah, the forward function is re-evaluated when we call `ctx.saved_tensors` again. 

The content of `storage` is not overwritten, because previous contents were removed in the last access, so they are newly added to the dict.

Reply:
Figured out how to do above and it is now tested.